src/test/java/core/basesyntax/RegistrationServiceTest.java

```diff
@@ -0,0 +1,177 @@
+package core.basesyntax;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import core.basesyntax.service.RegistrationException;
+import core.basesyntax.service.RegistrationService;
+import core.basesyntax.service.RegistrationServiceImpl;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class RegistrationServiceTest {
+    private static final String CORRECT_LOGIN = "1234567";
+    private static final String CORRECT_LOGIN_NEW = "7654321";
+    private static final String INCORRECT_LOGIN = "123";
+    private static final String BORDER_LOGIN = "123456";
+    private static final String EMPTY_LINE = "";
+    private static final String CORRECT_PASSWORD = "1234567";
+    private static final String INCORRECT_PASSWORD = "123";
+    private static final String BORDER_PASSWORD = "123456";
+    private static final int CORRECT_AGE = 22;
+    private static final int IN_CORRECT_AGE = 15;
+    private static final int BORDER_AGE = 18;
+    private static final int NEG_AGE = -1;
+    private static final User testUser = new User();
+    private static final User testUserNew = new User();
+    private static RegistrationService registrationService;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+        testUserNew.setLogin(CORRECT_LOGIN_NEW);
+        testUserNew.setPassword(CORRECT_PASSWORD);
+        testUserNew.setAge(CORRECT_AGE);
+    }
+
+    @BeforeEach
+    void setUp() {
+        testUser.setAge(CORRECT_AGE);
+        testUser.setPassword(CORRECT_PASSWORD);
+        testUser.setLogin(CORRECT_LOGIN);
+        Storage.people.clear();
+    }
+
+    @Test
+    void register_loginGreaterThanMinimum_ok() throws RegistrationException {
+        String login = CORRECT_LOGIN;
+        testUser.setLogin(login);
+        registrationService.register(testUser);
+        Assertions.assertEquals(login,testUser.getLogin());
+    }
+
+    @Test
+    void register_loginLesserThanMinimum_notOk() {
+        testUser.setLogin(INCORRECT_LOGIN);
+        Assertions.assertThrows(RegistrationException.class,
+                () -> registrationService.register(testUser));
+    }
+
+    @Test
+    void register_emptyLogin_notOk() {
+        testUser.setLogin(EMPTY_LINE);
+        Assertions.assertThrows(RegistrationException.class,
+                () -> registrationService.register(testUser));
+    }
+
+    @Test
+    void register_minimumLogin_ok() throws RegistrationException {
+        String login = BORDER_LOGIN;
+        testUser.setLogin(login);
```

Comment:
```suggestion
        testUser.setLogin(BORDER_LOGIN);
```
avoid creating redundant local variables where possible